File: NimdaSecurity/Greedy/Java/BOJ7568.java
import java.util.*;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;



public class Main {

    static class Pair{
        int first;
        int second;

        public Pair(int first, int second) {
            this.first = first;
            this.second = second;
        }
    }

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        ArrayList<Pair> info = new ArrayList();

        for (int i = 0; i < n; i++) {
            int weight = input.nextInt();
            int height = input.nextInt();
            info.add(new Pair(weight, height));
        }

        int[] rank = new int[n];
        for (int i = 0; i < n; i++) {
            int rank_number = 1;
            Pair standard_person = info.get(i);

            for (int j = 0; j < n; j++) {
                Pair compare_person = info.get(j);
                if (standard_person.first < compare_person.first &&standard_person.second < compare_person.second) {
                    rank_number++;
                }
            }

            rank[i] = rank_number;
        }

        for (int i = 0; i < n; i++) {
            System.out.print(rank[i] + " ");
        }


    }

}

File: NimdaSecurity/Greedy/C++/BOJ7568.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {

    ios::sync_with_stdio(false);
    cin.tie(NULL);

    int n;
    cin >> n;
    vector<pair<int, int>> info;
    int x, y;
    for (int i = 0; i < n; i++) {
        cin >> x >> y;
        info.push_back({x, y});
    }

    vector<int> rank(n, 1);
    for (int i = 0; i < n; i++) {
        int rank_number = 1;
        pair<int, int> standard_person = info[i];
        for (int j = 0; j < n; j++) {
            pair<int, int> compare_person = info[j];
            if (standard_person.first < compare_person.first && standard_person.second < compare_person.second) {
                rank_number++;
            }
        }
        rank[i] = rank_number;
    }

    for (int i = 0; i < n; i++)
        cout << rank[i] << " ";

}

